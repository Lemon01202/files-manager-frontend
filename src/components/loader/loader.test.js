import React from 'react';
import { render, screen } from '@testing-library/react';
import Loader from './Loader';

jest.mock('@mui/material', () => ({
  CircularProgress: () => <div data-testid="circular-progress" />,
  Box: ({ children, sx }) => (
    <div data-testid="box" style={sx}>
      {children}
    </div>
  ),
}));

describe('Loader', () => {
  it('renders CircularProgress', () => {
    render(<Loader />);

    const box = screen.getByTestId('box');
    expect(box).toBeInTheDocument();

    const circularProgress = screen.getByTestId('circular-progress');
    expect(circularProgress).toBeInTheDocument();
    expect(box).toContainElement(circularProgress);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
