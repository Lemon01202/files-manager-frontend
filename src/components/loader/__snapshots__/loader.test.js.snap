// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="box"
    style="display: flex; justify-content: center; align-items: center; height: 100vh;"
  >
    <div
      data-testid="circular-progress"
    />
  </div>
</DocumentFragment>
`;
